src/java.desktop/share/classes/java/awt/image/ColorConvertOp.java

```diff
@@ -808,7 +808,7 @@ private BufferedImage nonICCBIFilter(BufferedImage src,
                 idx = 0;
                 for (int x = 0; x < w; x++) {
                     pixel = srcRas.getDataElements(x, y, pixel);
-                    color = srcCM.getNormalizedComponents(pixel, color, 0);
+                    color = srcCM.getNormalizedComponents(pixel, null, 0);
```

Comment:
Since this is executed for each pixel, it will generate garbage equal to the image size. Maybe we can cleanly split the usage of the two vars here? Note that the bug only occurs when the source image does not use an ICC profile, but this change would increase garbage for both ICC and non-ICC profiles.

Reply:
You mean keep the original returned float[] color separate from the one that's later over-written.
I'd thought about that but I don't think it is worth it.
It can also be over-written at line 835. There's just too much state tracking needed to save
the GC a tiny bit of effort freeing small transient objects.
Also it already is re-initialised for each scanline. See line 807, so it isn't (quite) the same array for the entire image anyway.

Reply:
> It can also be over-written at line 835.

Interesting, is it possible that that line has the same bug?
`color = dstColorSpace.fromCIEXYZ(dstColor);`
Does dstColor always have the same number of components as CIEXYZ? 

Is the logic of using CIEXYZ for mix of non-/ICC source and non-/ICC destination actually correct?